qiskit/qobj/qasm_qobj.py

```diff
@@ -0,0 +1,778 @@
+                attributes of the object.
+        """
+        self._data = kwargs
+
+    def __getstate__(self):
+        return self._data
+
+    def __setstate__(self, state):
+        self._data = state
+
+    def __getattr__(self, attr):
+        try:
+            return self._data[attr]
+        except KeyError:
+            raise AttributeError
+
+    def __setattr__(self, name, value):
+        if not hasattr(self, name):
+            self._data[name] = value
+        else:
+            super().__setattr__(name, value)
+
+    def to_dict(self):
+        """Return a dict representation of the QASM Qobj experiment header.
+
+        Returns:
+            dict: The dictionary form of the QasmQobjExperimentHeader.
+        """
+        return self._data
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new QasmQobjExperimentHeader object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the experiment header
+
+        Returns:
+            QasmQobjExperimentHeader: The object from the input dictionary.
+        """
+        return cls(**data)
+
+    def __eq__(self, other):
+        return_val = False
+        if isinstance(other, QasmQobjExperimentHeader):
+            if self.to_dict() == other.to_dict():
+                return_val = True
+        return return_val
+
+
+class QasmQobjExperimentConfig:
+    """Configuration for a single QASM experiment in the qobj."""
+    _data = {}
+
+    def __init__(self, **kwargs):
+        """Instantiate a new QasmQobjExperimentConfig object.
+
+        Args:
+            kwargs: arbitrary keyword arguments that can be accessed as
+                attributes of the object.
+        """
+
+        self._data = kwargs
+
+    def __getstate__(self):
+        return self._data
+
+    def __setstate__(self, state):
+        self._data = state
+
+    def __getattr__(self, attr):
+        try:
+            return self._data[attr]
+        except KeyError:
+            raise AttributeError
+
+    def __setattr__(self, name, value):
+        if not hasattr(self, name):
+            self._data[name] = value
+        else:
+            super().__setattr__(name, value)
+
+    def to_dict(self):
+        """Return a dictionary format representation of the QASM Qobj.
+
+        Returns:
+            dict: The dictionary form of the QobjHeader.
+        """
+
+        return self._data
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new QasmQobjExperimentConfig object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the experiment config
+
+        Returns:
+            QasmQobjExperimentConfig: The object from the input dictionary.
+        """
+        return cls(**data)
+
+    def __eq__(self, other):
+        return_val = False
+        if isinstance(other, QasmQobjExperimentConfig):
+            if self.to_dict() == other.to_dict():
+                return_val = True
```

Comment:
`return True` for the `__eq__` methods followed by `return False` on the next line feels easier to read (this way I feel like I'm missing some nuanced logic but I don't see any).

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/42f1d26e78c5d67ad8def5d2717e460a9a9162b1